Wikipedia/Code/SavedArticlesFetcherProgressManager.swift

```diff
@@ -0,0 +1,63 @@
+/*
+ This class acts as a bridge between the SavedArticlesFetcher (which retrieves article html and images) and VC's which want to display save article fetching progress. It does so while being minimally invasive to SavedArticlesFetcher, observing only the SavedArticlesFetcher's 'fetchesInProcessCount'. Interested VC's can access the SavedArticlesFetcherProgressManager.shared.progress - but be aware that this "progress" object is not re-used, so you'll need to observe it to know when it gets re-set, and you'll then have to observe any properties you are interested in of the re-set progress. Reason:
+ 
+ "NSProgress objects cannot be reused. Once they’re done, they’re done. Once they’re cancelled, they’re cancelled. If you need to reuse an NSProgress, instead make a new instance and provide a mechanism so the client of your progress knows that the object has been replaced, like a notification." ( Source: https://developer.apple.com/videos/play/wwdc2015/232/ by way of https://stinkykitten.com/index.php/2017/08/13/nsprogress/ )
+*/
+
+@objcMembers class SavedArticlesFetcherProgressManager: NSObject, ProgressReporting {
+
+    static let shared = SavedArticlesFetcherProgressManager()
+
+    dynamic private(set) var progress = SavedArticlesFetcherProgressManager.newProgress()
+
+    private var fetchesInProcessCountObservation: NSKeyValueObservation?
+    
+    private static func newProgress() -> Progress {
+        return Progress.discreteProgress(totalUnitCount: -1)
+    }
+
+    private func resetProgress() {
+        // Reminder - see the note above for why this gets reset after finish.
+        progress = SavedArticlesFetcherProgressManager.newProgress()
+    }
+    
+    var fetcher: SavedArticlesFetcher? {
```

Comment:
Any reason not to init this class with the fetcher?

Reply:
Oops I'll try.